src/main/java/mods/eln/sixnode/wirelesssignal/WirelessUtils.kt

```diff
@@ -0,0 +1,246 @@
+package mods.eln.sixnode.wirelesssignal
+
+import mods.eln.misc.Coordonate
+import mods.eln.misc.Line3f
+import mods.eln.misc.Utils
+import mods.eln.misc.Vec3f
+import mods.eln.sixnode.wirelesssignal.tx.WirelessSignalTxElement
+import net.minecraft.block.Block
+import net.minecraft.block.material.Material
+import net.minecraft.init.Blocks
+import net.minecraft.server.MinecraftServer
+import net.minecraft.world.World
+import net.minecraft.world.chunk.Chunk
+
+import java.util.*
+import kotlin.collections.Map.Entry
+
+data class WirelessRaytraceKey(val startCoordonate: Coordonate, val endCoordonate: Coordonate)
+
+fun costOfBlockContaining(seg: Line3f, dim: Int) = seg.length() * when(seg.midpoint().toCoordonate(dim).block.material) {
+    Material.air -> 1.0
+    else -> 2.0
+}
+
+class WirelessCacheData(var segments: List<Line3f>, var dim: Int) {
+    var validTick: Int = 0
+    var cachedCost: Double = 0.0
+    var costMap: MutableMap<Chunk, Double> = HashMap()
+    var chunkMap: MutableMap<Chunk, MutableList<Line3f>> = HashMap()
+
+    init {
+        for(seg in segments) {
+            chunkMap.getOrPut(seg.midpoint().toCoordonate(dim).chunk, { -> ArrayList<Line3f>() }).add(seg)
+        }
+        populateCosts(true)
+    }
+
+    fun populateCosts(full: Boolean = false): Double {
+        var curTick = MinecraftServer.getServer().tickCounter
+        if(validTick == curTick)
+            return cachedCost
+        validTick = curTick
+
+        var total: Double = 0.0
+
+        for((chunk, segments) in chunkMap) {
+            if(full || chunk.isChunkLoaded) {
+                var chunk_total = 0.0
+                for(seg in segments) {
+                    chunk_total += costOfBlockContaining(seg, dim)
+                }
+                costMap.put(chunk, chunk_total)
+            }
+            total += costMap.getOrDefault(chunk, 0.0)
+        }
+
+        cachedCost = total
+        return total
+    }
+}
+
+
+object WirelessUtils {
```

Comment:
This object probably shouldn't exist; it seems to have no state whatsoever. Functions can exist outside of classes, and can still be private or internal. I'm not saying it's the only way, but give it a bit of thought.